### PR TITLE
remoteInterface merge; Local desktop socket support; Remote cameraMatrix support

### PR DESCRIPTION
**Pull request by hobinjk-ptc**
Mostly a merge of the remoteInterface branch from toolboxedgenet but also folds in some miscellaneous improvements I was working on locally

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -540,6 +549,7 @@
 <canvas class="Canvas" id="canvas"></canvas> <!--touch-action="none"-->
 <div class="Canvas" id="p5WebGL" style="position: absolute; left: 0; top: 0;"></div> <!--touch-action="none"-->
 <canvas id="glcanvas"></canvas>
+<canvas class="Canvas" id="testcan" style="position: absolute; left: 0; top: 0; transform: scale(0.5); display: none"></canvas> <!--touch-action="none"-->
```
Comment (benptc) [inline index.html:552]: Do we still need this? Looks like what it was used for may have been removed

File: src/network/index.js [hunk 2]
```diff
@@ -403,16 +508,28 @@ realityEditor.network.initializeDownloadedNode = function(objectKey, frameKey, n
...
 realityEditor.network.addHeartbeatObject = function (beat) {
-    if (beat.id) {
+    console.log('addHeartbeatObject', beat);
+    if (!realityEditor.device.loaded) {
+        console.log('addHeartbeatObject called before init done', beat);
+        setTimeout(() => {
+            realityEditor.network.addHeartbeatObject(beat);
+        }, 500);
+        return;
+    }
```
Comment (benptc) [inline src/network/index.js:518]: Nice safeguard 👍

**Review by benptc — APPROVED** [review]
Great changes!

File: index.html (hunk 1 above)
Comment (hobinjk-ptc) [reply index.html:552]: Nope, removed. Thanks for pointing this out!

File: src/network/realtime.js [hunk 3]
```diff
@@ -184,20 +194,21 @@ createNameSpace("realityEditor.network.realtime");
      */
     function setupServerSockets() {
         var ipList = [];
+        console.log("---sssss-w-w-w---- 1")
```
[the remark this answers is not in the document]
Comment (hobinjk-ptc) [reply src/network/realtime.js:197]: Good point! I'll scrub everything clean

File: src/network/index.js [hunk 4]
```diff
@@ -738,7 +855,7 @@ realityEditor.network.onAction = function (action) {
         }
 
         if (thisAction.reloadLink.object in objects) {
-            let urlEndpoint = 'http://' + objects[thisAction.reloadLink.object].ip + ':' + realityEditor.network.getPort(objects[thisAction.reloadLink.object]) + '/object/' + thisAction.reloadLink.object + '/frame/' +thisAction.reloadLink.frame;
+            let urlEndpoint = realityEditor.network.getURL(objects[thisAction.reloadLink.object].ip, realityEditor.network.getPort(objects[thisAction.reloadLink.object]), '/object/' + thisAction.reloadLink.object + '/frame/' +thisAction.reloadLink.frame);
```
[the remark this answers is not in the document]
Comment (hobinjk-ptc) [reply src/network/index.js:858]: Valentin did the heavy lifting for all of these rewrites, glad it wasn't me!

**Pull request merged**